src/web/WebSettingsService.cpp

```diff
@@ -88,90 +87,111 @@ void WebSettings::read(WebSettings & settings, JsonObject root) {
+        String nvs_boot = EMSESP::nvs_.getString("boot");
+        if (!nvs_boot.isEmpty()) {
+            settings.board_profile = nvs_boot;
         }
-        // check valid pins in this board profile
-        if (!System::is_valid_gpio(data[0], psram) || !System::is_valid_gpio(data[1], psram) || !System::is_valid_gpio(data[2], psram)
-            || !System::is_valid_gpio(data[3], psram) || !System::is_valid_gpio(data[4], psram) || !System::is_valid_gpio(data[6], psram)) {
-            settings.board_profile = ""; // reset to factory default
+    }
+
+    // load the board profile from the settings, if it's not "default"
+    std::vector<int8_t> data; //  // led, dallas, rx, tx, button, phy_type, eth_power, eth_phy_addr, eth_clock_mode
+    if (settings.board_profile != "default") {
+        if (System::load_board_profile(data, settings.board_profile.c_str())) {
+            if (settings.board_profile == "CUSTOM") { // read pins, fallback to S32 values
+                data = {(int8_t)(root["led_gpio"] | 2),
+                        (int8_t)(root["dallas_gpio"] | 18),
+                        (int8_t)(root["rx_gpio"] | 23),
+                        (int8_t)(root["tx_gpio"] | 5),
+                        (int8_t)(root["pbutton_gpio"] | 0),
+                        (int8_t)(root["phy_type"] | PHY_type::PHY_TYPE_NONE),
+                        (int8_t)(root["eth_power"] | 0),
+                        (int8_t)(root["eth_phy_addr"] | 0),
+                        (int8_t)(root["eth_clock_mode"] | 0)};
+            }
+            // check valid pins in this board profile
+            if (!System::is_valid_gpio(data[0], psram) || !System::is_valid_gpio(data[1], psram) || !System::is_valid_gpio(data[2], psram)
+                || !System::is_valid_gpio(data[3], psram) || !System::is_valid_gpio(data[4], psram) || !System::is_valid_gpio(data[6], psram)) {
+                settings.board_profile = "default"; // reset to factory default
+            }
+        } else {
+            settings.board_profile = "default"; // can't find profile, use "default"
         }
-    } else {
-        settings.board_profile = ""; // reset to factory default
     }
-    if (settings.board_profile == "") {
-        // unknown, check for NVS or scan for ethernet, use default E32/E32V2/S32
-        settings.board_profile = EMSESP::nvs_.getString("boot");
-        if (!System::load_board_profile(data, settings.board_profile.c_str())) {
+
+    // still don't have a valid board profile. Let's see if we can determine one
+    if (settings.board_profile == "default") {
+#ifdef EMSESP_DEBUG
+        EMSESP::logger().debug("Trying to detect board and set board profile...");
+#endif
+
 #if defined(EMSESP_STANDALONE)
-            settings.board_profile = "S32";
+        settings.board_profile = "S32";
 #elif CONFIG_IDF_TARGET_ESP32
-            if (settings.board_profile == "" && !psram) { // empty: new test for E32
+        // check for no PSRAM, could be a E32 or S32
```

Comment:
Ok, i think now i found where i get bootloops. If we have no -DEMSESP_BOARD_PROFILE in platformio.ini it goes to "default" and starts here the test. But the test sets only E32 or E32V2, not S32 otherwise, it leaves on "default". 
Set it here to "S32" and overwrite to E32/E32V2 if ETH found.

Reply:
good find. I thought I defaulted to S32, but will adjust. thanks for testing

Reply:
Hopefully made a fix. i don;t have a E32 so can't test

Reply:
Yes, this also works.